pkg/codegen/go/gen.go

```diff
@@ -1079,12 +1079,17 @@ func generatePackageContextMap(tool string, pkg *schema.Package, goInfo GoPackag
 	// In addition, if the optional property's type is itself an object type, we also need to generate pointer
 	// types corresponding to all of it's nested properties, as our accessor methods will lift `nil` into
 	// those nested types.
-	var markOptionalPropertyTypesAsRequiringPtr func(props []*schema.Property, parentOptional bool)
-	markOptionalPropertyTypesAsRequiringPtr = func(props []*schema.Property, parentOptional bool) {
+	var markOptionalPropertyTypesAsRequiringPtr func(parent *schema.ObjectType, props []*schema.Property, parentOptional bool)
+	markOptionalPropertyTypesAsRequiringPtr = func(parent *schema.ObjectType, props []*schema.Property, parentOptional bool) {
 		for _, p := range props {
 			if obj, ok := p.Type.(*schema.ObjectType); ok && (!p.IsRequired || parentOptional) {
+				// Skip recursive recursive types. For example, JSONSchemaProps in the Kubernetes
+				// package can have properties with a type of JSONSchemaProps.
+				if parent != nil && parent.Token == obj.Token {
+					continue
+				}
```

Comment:
This is probably fine - but it might be more "correct" to pass a `seen` map into this recursive function and then check that.  That would ensure that if the recursion is two layers deep, it still gets caught.